crates/interpreter/src/interpreter/stack.rs

```diff
@@ -7,7 +7,7 @@ use std::vec::Vec;
 pub const STACK_LIMIT: usize = 1024;
 
 /// EVM stack with [STACK_LIMIT] capacity of words.
-#[derive(Debug, PartialEq, Eq, Hash)]
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
```

Comment:
this is problematic, because breaks the invariant that this has type hash STACK_LIMIT capacity

Reply:
yeah shouldn't derive but Self::new with copy, pls also add a comment

Reply:
How about we implement the clone manually, then we can use `clone()` as usual:

```rust
impl Clone for Stack {
    fn clone(&self) -> Self {
        let mut new_data = Vec::with_capacity(STACK_LIMIT);
        new_data.extend_from_slice(&self.data);
        Self { data: new_data }
    }
}
```

Reply:
that's what I mean

Reply:
@DaniPopes applied, ptal